src/_guides/language/language-tour.md

```diff
@@ -3730,56 +3780,29 @@ Note that the function's body doesn't need to use the Future API.
 Dart creates the Future object if necessary.
 
 
-<a id="await"></a>
-### Using await expressions with Futures
-
-An await expression has the following form:
-
-<pre>
-<b>await</b> <em>expression</em>
-</pre>
-
-You can use `await` multiple times in an async function.
-For example, the following code waits three times
-for the results of functions:
-
-<?code-excerpt "misc/lib/language_tour/async.dart (repeated-await)"?>
-{% prettify dart %}
-var entrypoint = await findEntrypoint();
-var exitCode = await runExecutable(entrypoint, args);
-await flushThenExit(exitCode);
-{% endprettify %}
-
-In <code>await <em>expression</em></code>,
-the value of <code><em>expression</em></code> is usually a Future;
-if it isn't, then the value is automatically wrapped in a Future.
-This Future object indicates a promise to return an object.
-The value of <code>await <em>expression</em></code> is that returned object.
-The await expression makes execution pause until that object is available.
-
-**If `await` doesn't work, make sure it's in an async function.**
-For example, to use `await` in your app's `main()` function,
-the body of `main()` must be marked as `async`:
+<a id="await-for"></a>
+### Handling Streams
 
-<?code-excerpt "misc/lib/language_tour/async.dart (main)"?>
-{% prettify dart %}
-Future main() async {
-  checkVersion();
-  print('In main: version is ${await lookUpVersion()}');
-}
-{% endprettify %}
+When you need to get values from a **Stream**,
+you have two options:
 
+* Use `async` and an _asynchronous for loop_ (`await for`).
+* Use the [Stream API](/guides/libraries/library-tour#stream).
 
-<a id="await-for"></a>
-### Using asynchronous for loops with Streams
+<aside class="alert alert-warning" markdown="1">
+**Note:**
+Asynchronous for loops aren't always appropriate.
+Use them only if it's OK for your program to do nothing
+while it waits for values from a single stream.
+</aside>
 
 An asynchronous for loop has the following form:
 
-<pre>
-<b>await for</b> (<em>variable declaration</em> in <em>expression</em>) {
+```dart
```

Comment:
I started doing that, then realized I'd need a real stream, and that seemed like too much trouble for code that doesn't actually do anything.

Reply:
Since you won't be testing the code, but only subjecting to the analyzer, then you don't need to create stream; e.g. something like this could be added to the end of `examples/misc/lib/language_tour/async.dart`:

```dart
  (Stream expression) async {
    await for (var variableDeclaration in expression) {
      // Executes each time the stream emits a value.
    }
  };
```

**Note** the added `var` keyword.